dandi/download.py

```diff
@@ -358,13 +358,34 @@ def download_generator(self) -> Iterator[dict]:
                         lock=lock,
                     )
 
+                def _progress_filter(gen):
+                    """To reduce load on pyout etc, make progress reports only if enough time
+                    from prior report has passed (over 2 seconds) or we are done (got 100%).
+
+                    Note that it requires "awareness" from the code below to issue other messages
+                    with bundling with done% reporting if reporting on progress of some kind (e.g.,
+                    adjusting "message").
+                    """
+                    prior_time = 0
+                    warned = False
+                    for rec in gen:
+                        current_time = time.time()
+                        if done_perc := rec.get("done%", 0):
+                            if isinstance(done_perc, (int, float)):
+                                if current_time - prior_time < 2 and done_perc != 100:
+                                    continue
+                            elif not warned:
```

Comment:
Don't you want to set `warned = True` here?

Reply:
thanks! will do